### PR TITLE
Falling back to study cancer type when no cancer type clinical data exists

### PR DESCRIPTION
**Pull request by onursumer**
# What? Why?
Fix https://github.com/cBioPortal/cbioportal/issues/2863.

Changes proposed in this pull request:
- Now using the study cancer type for samples that don't have cancer type clinical data

# Checks
- [ ] Follows [7 rules of great commit messages](http://chris.beams.io/posts/git-commit/). For most PRs a single commit should suffice, in some cases multiple topical commits can be useful. During review it is ok to see tiny commits (e.g. Fix reviewer comments), but right before the code gets merged to master or rc branch, any such commits should be squashed since they are useless to the other developers. Definitely avoid [merge commits, use rebase instead.](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)
- [ ] Follows the [Airbnb React/JSX Style guide](https://github.com/airbnb/javascript/tree/master/react).
- [ ] Make sure your commit messages end with a Signed-off-by string (this line
  can be automatically added by git if you run the `git-commit` command with
  the `-s` option)

### CODE REVIEW

File: src/pages/resultsView/ResultsViewPageStore.ts [hunk 1]
```diff
@@ -106,6 +109,21 @@ export class ResultsViewPageStore {
         invoke: async () => fetchSamples(this.sampleIds, this.studyId)
     }, []);
 
+    readonly samplesWithoutCancerTypeClinicalData = remoteData({
+        await: () => [
+            this.sampleIds,
+            this.clinicalDataForSamples
+        ],
+        invoke: async () => fetchSamplesWithoutCancerTypeClinicalData(this.sampleIds, this.studyId, this.clinicalDataForSamples)
```
Comment (onursumer) [inline src/pages/resultsView/ResultsViewPageStore.ts]: here we still depend on the store's study id to fetch the samples...

File: src/pages/resultsView/mutation/MutationMapper.tsx [hunk 2]
```diff
@@ -122,8 +122,14 @@ export default class MutationMapper extends React.Component<IMutationMapperProps
                                     </span>
                                 </div>)
                             }
-                            <LoadingIndicator isLoading={this.props.store.clinicalDataForSamples.isPending} />
-                            {!this.props.store.clinicalDataForSamples.isPending && (
+                            <LoadingIndicator
+                                isLoading={
+                                    this.props.store.clinicalDataForSamples.isPending ||
```
Comment (alisman) [inline src/pages/resultsView/mutation/MutationMapper.tsx]: we should probably hide this fallback thing.  Like this wrap this in another remoteData.  It's just too much to grock.

File: src/pages/resultsView/ResultsViewPageStore.ts (hunk 1 above)
Comment (adamabeshouse) [inline src/pages/resultsView/ResultsViewPageStore.ts]: is that a problem?

Comment (onursumer) [inline src/pages/resultsView/ResultsViewPageStore.ts]: not for now, but we will need to redesign it when we want to support multiple studies

Comment (adamabeshouse) [inline src/pages/resultsView/ResultsViewPageStore.ts]: true. I say just leave it and we'll make those changes all at once

**Pull request merged**